app/models/spree/ability_decorator.rb

```diff
@@ -283,6 +283,16 @@ def add_relationship_management_abilities(user)
       user.enterprises.include? enterprise_relationship.parent
     end
   end
+
+  def add_enterprise_fee_summary_abilities(user)
+    feature_enabled = FeatureFlags.new(user).enterprise_fee_summary_enabled?
+    return unless feature_enabled
+
+    # Reveal the report link in spree/admin/reports#index
+    can [:enterprise_fee_summary], :report
+    # Allow direct access to the report resource
+    can [:admin, :new, :create], :enterprise_fee_summary
+  end
```

Comment:
This file is already quite unmanageable and although I love the path you take @kristinalim I'd go one step further and split abilities into multiple files. It is as easy as registering the individual classes as we do in line 298. If you agree, could you create a tech debt issue for it?

Reply:
Nice! I created the tech debt issue: #3345